CMakeLists.txt

```diff
@@ -470,7 +470,7 @@ if (S2N_INTERN_LIBCRYPTO)
           mkdir s2n_libcrypto &&
           cd s2n_libcrypto &&
           # extract libcrypto objects from the archive
-          ar x ../s2n_libcrypto.a &&
+          gcc-ar x ../s2n_libcrypto.a &&
```

Comment:
We want to use `CMAKE_AR` so we inherit the configured ar program.
```suggestion
          ${CMAKE_AR} x ../s2n_libcrypto.a &&
```

We also need to make sure that variable is set, even if the user doesn't configure it. If not, we'll need to fallback to `ar`.